src/core/coap/coap_secure.cpp

```diff
@@ -320,6 +320,11 @@ CoapSecure &CoapSecure::GetOwner(const Context &aContext)
     return coap;
 }
 
+void CoapSecure::HandleRetransmissionTimer(Timer &aTimer)
+{
+    GetOwner(aTimer).Coap::HandleRetransmissionTimer();
```

Comment:
I remembered we had the same situation earlier in `DatasetManager()` and its two sub-classes `PendingDatasetBase()` and `ActiveDatasetBase` and actually used the exact same approach by having the timer handler method to be passed to base class.